lib/api/controllers/documentController.js

```diff
@@ -362,26 +418,27 @@ class DocumentController extends BaseController {
    * @param {Request} request
    * @returns {Promise<Object>}
    */
-  mDelete(request) {
-    assertHasBody(request);
-    assertBodyHasAttribute(request, 'ids');
-    assertBodyAttributeType(request, 'ids', 'array');
-    assertHasIndexAndCollection(request);
-
-    let response;
-
-    return this.engine.mdelete(request)
-      .then(_response => {
-        response = _response;
+  mDelete (request) {
+    const
+      deletedIds = [],
+      ids = this.getArrayParam(request, 'body.ids'),
+      refresh = this.getStringParam(request, 'args.refresh', 'false'),
+      { index, collection } = this.getIndexAndCollection(request);
+
+    return this.publicStorage.mDelete(index, collection, ids, { refresh })
+      .then(({ documents, errors }) => {
+        for (let i = 0; i < documents.length; i++) {
+          deletedIds.push(documents[i]._id);
+        }
 
-        if (response.error.length > 0) {
-          request.setError(errorsManager.getError('deletion_failed', response.error));
+        if (errors.length > 0) {
+          request.setError(errorsManager.getError('deletion_failed', errors));
         }
 
-        return this.kuzzle.notifier.notifyDocumentMDelete(
-          request, response.result);
+        // @todo should be done in clientAdapter ?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)